addons/website/static/tests/tours/grid_layout.js

```diff
@@ -46,3 +46,41 @@ wTourUtils.registerWebsitePreviewTour('website_replace_grid_image', {
     },
     ...wTourUtils.clickOnSave()
 ]);
+
+wTourUtils.registerWebsitePreviewTour("scroll_to_new_grid_item", {
+    test: true,
+    url: "/",
+    edition: true,
+}, () => [
+    // Drop enough snippets to scroll.
+    wTourUtils.dragNDrop({id: "s_text_image", name: "Text - Image"}),
+    wTourUtils.dragNDrop({id: "s_image_text", name: "Image - Text"}),
+    wTourUtils.dragNDrop({id: "s_image_text", name: "Image - Text"}),
+    // Toggle the first snippet to grid mode.
+    wTourUtils.clickOnSnippet({id: "s_text_image", name: "Text - Image"}),
+    wTourUtils.changeOption("layout_column", 'we-button[data-name="grid_mode"]'),
+    // Add a new grid item.
+    wTourUtils.changeOption("layout_column", 'we-button[data-add-element="image"]'),
+    {
+        content: "Select the new image in the media dialog",
+        trigger: '.o_select_media_dialog img[title="s_banner_default_image.jpg"]',
+    }, {
+        content: "Check that the page scrolled to the new grid item",
+        trigger: ":iframe .s_text_image .o_grid_item:nth-child(3)",
+        run: function () {
+            // Leave some time to the page to scroll.
+            setTimeout(() => {
+                const newItemPosition = this.anchor.getBoundingClientRect();
+                if (newItemPosition.top < 0) {
+                    console.error("The page did not scroll to the new grid item.");
+                }
+                document.body.classList.add("o_scrolled_to_grid_item");
+            }, 500);
+        },
+    }, {
+        content: "Make sure the scroll check is done",
+        trigger: ".o_scrolled_to_grid_item",
+        isCheck: true,
+    },
+    ...wTourUtils.clickOnSave(),
```

Comment:
Why you added back the click on save @sobo-odoo ? :thinking:

Reply:
Because when an image is added/uploaded near the end of a tour, the test can fail sometimes (there is a message indicating that there is no savepoint) :slightly_smiling_face: 
It happens when the GETs of the image are not completely done when the test is over, making the test fail (while the tour is working)
I encountered it before and saving at the end of the tour solves the issue, as saving waits for all the request to be done so the test always ends in a consistent state 

I had the issue in [this PR](https://github.com/odoo/odoo/pull/142280#pullrequestreview-1746226457) (I can show you the discord conversation about this if you want) :wink: